source/renderer/app/components/staking/stake-pools/StakePoolsRanking.js

```diff
@@ -0,0 +1,316 @@
+  },
+  actionLearnMore: {
+    id: 'staking.stakePools.learnMore',
+    defaultMessage: '!!!Learn more',
+    description: 'Learn more action of ranking panel.',
+  },
+});
+
+const learnMoreUrlMap = {
+  'en-US': 'https://iohk.zendesk.com/hc/en-us',
+  'ja-JP': 'https://iohk.zendesk.com/hc/ja',
+};
+
+const walletSelectorLanguageMap = {
+  'en-US': 'en-selector',
+  'ja-JP': 'ja-selector',
+};
+
+type Props = {
+  wallets: Array<Wallet>,
+  currentLocale: string,
+  onOpenExternalLink: Function,
+  onRank: Function,
+  isLoading: boolean,
+};
+
+type State = {
+  selectedWalletId: ?string,
+  sliderValue: number,
+};
+
+const OUT_OF_RANGE_MAX_AMOUNT = new BigNumber('11000000000');
+const MIN_AMOUNT = new BigNumber('10');
+const MAX_AMOUNT = new BigNumber('44000000');
+
+@observer
+export default class StakePoolsRanking extends Component<Props, State> {
+  static contextTypes = {
+    intl: intlShape.isRequired,
+  };
+
+  static defaultProps = {
+    wallets: [],
+  };
+
+  state = {
+    selectedWalletId: null,
+    sliderValue: MIN_AMOUNT.toNumber(),
+  };
+
+  componentDidMount() {
+    this.onSelectedWalletChange('0');
+  }
+
+  getAllAvailableAmount = () => {
+    const { wallets } = this.props;
+
+    if (wallets.length) {
+      return wallets
+        .map((w: Wallet) => w.availableAmount)
+        .reduce(
+          (acc: BigNumber, cur: BigNumber) => acc.plus(cur),
+          new BigNumber(0)
+        );
+    }
+
+    return MIN_AMOUNT;
+  };
+
+  onSelectedWalletChange = (selectedWalletId: string) => {
+    const { wallets } = this.props;
+    const selectedWallet = wallets.find(
+      wallet => wallet.id === selectedWalletId
+    );
+    let sliderValue = null;
+
+    if (selectedWalletId === '0') {
+      sliderValue = Math.min(
+        Math.floor(this.getAllAvailableAmount().toNumber()),
+        MAX_AMOUNT.toNumber()
+      );
+    } else if (selectedWallet) {
+      sliderValue = Math.floor(selectedWallet.availableAmount.toNumber());
+    } else {
+      sliderValue = MIN_AMOUNT.toNumber();
+    }
+
+    this.setState({
+      selectedWalletId,
+      sliderValue: Math.max(sliderValue, MIN_AMOUNT.toNumber()),
+    });
+  };
+
+  onSliderChange = (sliderValue: number) => {
+    this.props.onRank(sliderValue);
+    this.setState({ sliderValue, selectedWalletId: null });
+  };
+
+  generateInfo = () => {
+    const { intl } = this.context;
+    const { wallets, currentLocale } = this.props;
+    const { selectedWalletId, sliderValue } = this.state;
+    const allWalletsItem = {
+      label: intl.formatMessage(messages.rankingAllWallets),
+      value: '0',
+    };
+    const walletSelectorOptions = [
+      allWalletsItem,
+      ...wallets.map((w: Wallet) => ({
+        label: w.name,
+        value: w.id,
+      })),
+    ];
+    const walletSelectorClasses = classnames([
+      styles.walletSelector,
+      walletSelectorLanguageMap[currentLocale],
+      selectedWalletId === null ? 'no-value-selected' : null,
```

Comment:
@yakovkaravelov transform classes to camel case `no-value-selected` to `noValueSelected`

Reply:
👍